app/src/main/java/org/mozilla/fenix/home/WallpapersObserver.kt

```diff
@@ -0,0 +1,96 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.home
+
+import android.widget.ImageView
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.isVisible
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.Store
+import org.mozilla.fenix.components.AppStore
+import org.mozilla.fenix.components.appstate.AppAction
+import org.mozilla.fenix.components.appstate.AppState
+import org.mozilla.fenix.ext.scaleToBottomOfView
+import org.mozilla.fenix.wallpapers.Wallpaper
+import org.mozilla.fenix.wallpapers.WallpapersUseCases
+
+/**
+ * [LifecycleObserver] that will immediately start observing the store for wallpapers updates
+ * to apply them to the passed in [wallpaperImageView] and automatically stop observing for updates
+ * when the [LifecycleOwner] is destroyed.
+ *
+ * @param appStore Holds the details abut the current wallpaper.
+ * @param wallpapersUseCases Used for interacting with the wallpaper feature.
+ * @param wallpaperImageView Serves as the target when applying wallpapers.
+ */
+class WallpapersObserver(
```

Comment:
In composables we normally use `observeAsComposableState` meaning [`observe(lifecycleOwner)`](https://github.com/mozilla-mobile/android-components/blob/6c0fe91ab83614265f533b819bf406fc046b2227/components/lib/state/src/main/java/mozilla/components/lib/state/ext/ComposeExtensions.kt#L37) meaning again that we start listening for updates after the screen is already displayed so in a quick compose implementation we'd indeed have the same issue.
The only code that matters in the observer is [the `observeManually`](https://github.com/mozilla-mobile/fenix/blob/642841a6f091749650996b5b158854fdc659522a/app/src/main/java/org/mozilla/fenix/home/WallpapersObserver.kt#L64-L76) call, the rest being added here just to have a cleaner separation of concerns and avoid adding coroutine scopes and subsciptions to HomeFragment.
So we'll indeed need a solution for Compose where instead of using the Fragment as a LifecycleOwner to observe maybe we could use the ComposeView's attached/detached state.